From: Wengang Wang <wen.gang.wang@oracle.com>
Date: Wed, 8 Dec 2010 20:34:39 +0800
Subject: [PATCH] ocfs2/dlm: make existing convertion precedent over
 new lock
Patch-mainline: yes
Git-commit: 66f4500573fe5a1b455e5f7b30068a623a94117f

Make existing convertion precedent over new lock. It makes o2dlm locking more
like fair locking.

Signed-off-by: Wengang Wang <wen.gang.wang@oracle.com>
Signed-off-by: Joel Becker <joel.becker@oracle.com>
Acked-by: David Sterba <dsterba@suse.cz>
---
 fs/ocfs2/dlm/dlmlock.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/fs/ocfs2/dlm/dlmlock.c
+++ b/fs/ocfs2/dlm/dlmlock.c
@@ -106,6 +106,9 @@ static int dlm_can_grant_new_lock(struct
 
 		if (!dlm_lock_compatible(tmplock->ml.type, lock->ml.type))
 			return 0;
+		if (!dlm_lock_compatible(tmplock->ml.convert_type,
+					 lock->ml.type))
+			return 0;
 	}
 
 	return 1;
